src/rust/engine/src/core.rs

```diff
@@ -84,7 +84,7 @@ impl<'x> Params {
       .binary_search_by(|probe| probe.type_id().cmp(&type_id))
   }
 
-  pub fn type_ids<'a>(&'a self) -> impl Iterator<Item = TypeId> + 'a {
+  pub fn type_ids(&self) -> impl Iterator<Item = TypeId> + '_ {
```

Comment:
The clippy lint that was triggered:
```
error: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
  --> src/core.rs:87:3
   |
87 |   pub fn type_ids<'a>(&'a self) -> impl Iterator<Item = TypeId> + 'a {
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/lib.rs:7:3
   |
7  |   clippy::all,
   |   ^^^^^^^^^^^
   = note: `#[deny(clippy::needless_lifetimes)]` implied by `#[deny(clippy::all)]`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes

error: aborting due to previous error

error: could not compile `engine`

To learn more, run the command again with --verbose.
```

Just removing the lifetime was insufficient to fix:

```
error[E0759]: `self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
  --> src/core.rs:88:12
   |
87 |   pub fn type_ids(&self) -> impl Iterator<Item = TypeId> {
   |                   ----- this data with an anonymous lifetime `'_`...
88 |     self.0.iter().map(|k| *k.type_id())
   |     ------ ^^^^
   |     |
   |     ...is captured here...
   |
note: ...and is required to live as long as `'static` here
  --> src/core.rs:87:29
   |
87 |   pub fn type_ids(&self) -> impl Iterator<Item = TypeId> {
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to declare that the `impl Trait` captures data from argument `self`, you can add an explicit `'_` lifetime bound
   |
87 |   pub fn type_ids(&self) -> impl Iterator<Item = TypeId> + '_ {
   |                                                          ^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0759`.
error: could not compile `engine`
```